includes/class-wc-gateway-ppec-client.php

```diff
@@ -698,6 +699,59 @@ protected function _get_details_from_order( $order_id ) {
 		return $details;
 	}
 
+	/**
+	 * Get PayPal shipping address from customer.
+	 *
+	 * @return array Address
+	 */
+	protected function _get_address_from_customer() {
+		$customer = WC()->customer;
+
+		$shipping_address = new PayPal_Address;
+
+		$old_wc = version_compare( WC_VERSION, '3.0', '<' );
+
+		if ( $customer->get_shipping_address() || $customer->get_shipping_address_2() ) {
+			$shipping_first_name = $old_wc ? '' : $customer->get_shipping_first_name();
+			$shipping_last_name  = $old_wc ? '' : $customer->get_shipping_last_name();
+			$shipping_address_1  = $customer->get_shipping_address();
+			$shipping_address_2  = $customer->get_shipping_address_2();
+			$shipping_city       = $customer->get_shipping_city();
+			$shipping_state      = $customer->get_shipping_state();
+			$shipping_postcode   = $customer->get_shipping_postcode();
+			$shipping_country    = $customer->get_shipping_country();
+		} else {
+			// Fallback to billing in case no shipping methods are set. The address returned from PayPal
+			// will be stored in the order as billing.
+			$shipping_first_name = $old_wc ? ''                         : $customer->get_billing_first_name();
+			$shipping_last_name  = $old_wc ? ''                         : $customer->get_billing_last_name();
+			$shipping_address_1  = $old_wc ? $customer->get_address()   : $customer->get_billing_address_1();
+			$shipping_address_2  = $old_wc ? $customer->get_address_2() : $customer->get_billing_address_2();
+			$shipping_city       = $old_wc ? $customer->get_city()      : $customer->get_billing_city();
+			$shipping_state      = $old_wc ? $customer->get_state()     : $customer->get_billing_state();
+			$shipping_postcode   = $old_wc ? $customer->get_postcode()  : $customer->get_billing_postcode();
+			$shipping_country    = $old_wc ? $customer->get_country()   : $customer->get_billing_country();
+		}
+
+		$shipping_address->setName( $shipping_first_name . ' ' . $shipping_last_name );
+		$shipping_address->setStreet1( $shipping_address_1 );
+		$shipping_address->setStreet2( $shipping_address_2 );
+		$shipping_address->setCity( $shipping_city );
+		$shipping_address->setState( $shipping_state );
+		$shipping_address->setZip( $shipping_postcode );
+
+		// In case merchant only expects domestic shipping and hides shipping
+		// country, fallback to base country.
+		//
+		// @see https://github.com/woothemes/woocommerce-gateway-paypal-express-checkout/issues/139
+		if ( empty( $shipping_country ) ) {
```

Comment:
Can we more explicitly check for the scenario where the merchant is only shipping domestically? At first glance it seems like we might be reading too much into an empty shipping country.. would we hit this case if the form wasn't completely filled out?

In the WP.org thread linked in the issue ([#](https://wordpress.org/support/topic/an-error-occured-while-processing-your-paypal-payment?replies=6)), the "sell to" countries setting is referenced. Perhaps we can check that and the "ship to" countries settings to determine the proper action?